modal/functions.py

```diff
@@ -178,6 +210,21 @@ async def pop_function_call_outputs(
                     # return the last response to check for state of num_unfinished_inputs
                     return response
 
+    async def retry_input(self) -> None:
+        if not self._function_call_jwt or not self._input_id or not self._item:
+            raise ValueError("_function_call_jwt, _response_item, and _item must not be empty when retrying an input")
+
+        item = api_pb2.FunctionRetryInputsItem(idx=self._item.idx, input_id=self._input_id, input=self._item.input)
+        request = api_pb2.FunctionRetryInputsRequest(function_call_jwt=self._function_call_jwt, inputs=[item])
```

Comment:
I might be wrong, but I think the `input_id` must be part of the JWT for true validation.